module/validation/receipt_validator.go

```diff
@@ -0,0 +1,182 @@
+package validation
+
+import (
+	"errors"
+	"fmt"
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/encoding"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/signature"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+)
+
+type receiptValidator struct {
```

Comment:
I think we generally export implementations of interfaces, and return their specific type at the constructor:
```golang
func NewReceiptValidator(state protocol.ReadOnlyState, index storage.Index, results storage.ExecutionResults) receiptValidator
``` 
(just to have a consistent style in the code base)